src/lib.rs

```diff
@@ -54,6 +54,25 @@
 //!
 //! - Note: You can also use [`bms::parse::model::Bms::from_token_stream_with_ast`] to skip the AST building & parsing step.
 //!
+//! ## AST Extraction Usage
+//!
+//! You can also extract tokens back from an AST using the [`bms::ast::AstRoot::extract`] method, which serves as the inverse of [`bms::ast::AstRoot::from_token_stream`]:
+//!
+//! ```rust
+//! use bms_rs::bms::prelude::*;
+//!
+//! let source = std::fs::read_to_string("tests/files/lilith_mx.bms").unwrap();
+//! let LexOutput { tokens, lex_warnings } = TokenStream::parse_lex(&source);
+//! assert_eq!(lex_warnings, vec![]);
+//!
+//! // Build AST from tokens
+//! let AstBuildOutput { root, ast_build_warnings } = AstRoot::from_token_stream(&tokens);
+//! assert_eq!(ast_build_warnings, vec![]);
+//!
+//! // Extract tokens back from AST (inverse operation)
```

Comment:
```suggestion
//! // Extract tokens back from AST
```
Hmm, strictly speaking, it is not *inverse* operation. I think it's better to remove it.